docarray/documents/text.py

```diff
@@ -65,6 +65,22 @@ class MultiModalDoc(BaseDocument):
             text_doc=Text(text="hello world, how are you doing?"),
         )
         mmdoc.text_doc.text = mmdoc.text_doc.url.load()
+
+    This Document can be compared against another Document of the same type or a string.
+    When compared against another object of the same type, the pydantic BaseModel
+    equality check will apply which checks the equality of every attribute,
+    including `id`. When compared against a str, it will check the equality
+    of the `text` attribute against the given string.
+
+    .. code-block:: python
+
+        from docarray.documents Text
+
+        doc = Text(text='This is the main text', url='exampleurl.com')
+        doc2 = Text(text='This is the main text', url='exampleurl.com')
+
+        doc == 'This is the main text' # True
+        doc == doc2 # False, their ids are not equivalent
```

Comment:
isn't this a problem btw ? I would expect it to be true even thouhg id are sot similar

Reply:
I think this can be discussed in another PR, I would separate the concern, for now I just added tests to show explicitly the behavior